config/notifiers.go

```diff
@@ -521,6 +528,49 @@ func (c *SlackConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+// IncidentioConfig configures notifications via incident.io.
+type IncidentioConfig struct {
```

Comment:
What do you think about adding a field `Metadata` to this struct? Something similar to the [`Details` field](https://github.com/prometheus/alertmanager/blob/main/config/notifiers.go#L612) in the `OpsgenieConfig` struct.
This would enable users to define additional data (see [incident.io API definition](https://api-docs.incident.io/tag/Alert-Events-V2#operation/Alert%20Events%20V2_CreateHTTP!path=metadata&t=request) for reference).